src/util/tab-events.test.js

```diff
@@ -0,0 +1,185 @@
+/* eslint-env jest */
+/* eslint import/namespace: "off" */
+
+import { whenPageDOMLoaded, whenPageLoadComplete, whenTabActive } from 'src/util/tab-events'
+import * as eventToPromise from 'src/util/event-to-promise'
+
+describe('whenPageDOMLoaded', () => {
+    const tabId = 1
+
+    beforeAll(() => {
+        browser.webNavigation = {
+            onCommitted: jest.fn(),
+        }
+        eventToPromise.default = jest.fn().mockReturnValue(
+            new Promise((resolve, reject) => {
+                resolve()
+            })
+        )
+    })
+
+    test('should call the browser event to execute script and resolve promise if script executes', async () => {
+        browser.tabs = {
+            executeScript: jest.fn().mockReturnValue(
+                new Promise((resolve, reject) => {
+                    resolve()
+                })
+            ),
+        }
+        await whenPageDOMLoaded({tabId})
+        expect(browser.tabs.executeScript).toHaveBeenCalledWith(tabId, {code: 'undefined', runAt: 'document_end'})
+    })
+
+    test('should reject the promise if the script is not executed', async () => {
```

Comment:
Good catch. The test it seems was falsely passing all this time. Although I can't seem to figure out why it doesn't work. Any suggestions?